RELEASE.md

```diff
@@ -0,0 +1,135 @@
+- The Kanister release job will publish a new tag, update documentation, build all the docker images, and push them to the [ghcr.io](https://github.com/orgs/kanisterio/packages) registry.
+- Once the job is complete, a Slack notification will be sent to the kanister channel.
+- Post release announcement in kanister slack and https://groups.google.com/g/kanisterio
+
+### Pre-release workflow
+
+`pre-release` workflow serves to create a new version of kanister, it updates version number and creates a PR in kanister repo.
+
+The workflow can be triggered using workflow dispatch from the `Actions` tab in the repo: https://github.com/kanisterio/kanister/actions/workflows/pre-release.yml
+It has a required input of `release_tag`, which should be set to a version next to the current version.
+
+This will result in creating a PR with a version bump, like https://github.com/kanisterio/kanister/pull/2629
+**NOTE** PR description would look like `pre-release: Update version to ...`
+
+This PR would update the older Kanister version reference in various files.
+
+### Release workflow
+
+`release` workflow tags the repo using `release_tag` variable either from the merged pre-release PR or from workflow dispatch.
+It then uses the `goreleaser` tool to produce a github release and core images such as `ghcr.io/kanisterio/controller`, `ghcr.io/kanisterio/repo-server-controller`, `ghcr.io/kanisterio/kanister-tools` and `ghcr.io/kanisterio/kanister-kubectl-1.18`
+
+Published release artifacts include helm chart for the operator.
+
+Release workflow also builds and publishes docs and helm chart index into GH pages.
+
+### Release notes
+
+When working on pre-release PR the person doing the release should check that merged PRs have release notes when necessary.
+
+It can be checked by running a diff with previous release and looking at the PR commits:
+
+```
+git log 0.106.0..HEAD --invert-grep --grep='deps' --grep='docs' --grep='build' | grep -oh '^.*\(#[0-9]*\)'
+```
+
+And verifying that PRs have release notes.
+
+This is a bit tedious, but should be less of an issue as we establish a process of adding release notes.
+
+
+## Handling Failures
+
+### Retry
+
+Depending on where the release failed, some of the following steps may have succeeded:
+
+- Creation of a new tag in the kanister repo
+- PR raised in kanister to update kanister tools
+
+Before rerunning the release job manually, do the following:
+
+- Delete the tag manually from GitHub
+- Close the PRs created (to avoid confusion)
+
+## Manual Release (Only needed when automation is broken)
+
+Prerequisites:
+
+- Make sure you have admin access or access to push tags on Kanister
+- Make sure you have access to [push images to GHCR](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)
+
+### Create pre-release
+
+If pre-release pipeline does not work.
+
+```
+$ export PREV_TAG="0.42.0"
```

Comment:
If I didn't misread things, when the pre-release happens using actions we documented above that we are supposed to give the new version that we want to release kanister with. For example if the latest tag in kanister repo is `0.101.0` we are supposed to give `0.102.0`.
But here we are saying we have to give the previous tag? If that is the case, it might be confusing.

Reply:
Added clarification